xdsl/dialects/builtin.py

```diff
@@ -2102,6 +2102,14 @@ def shape_is_complete(self) -> bool:
         # Product of dimensions needs to equal length
         return n == len(self)
 
+    def verify(self) -> None:
+        # zero rank type should only hold 1 value
+        data_len = len(self.get_values())
+        if not self.type.get_shape() and data_len != 1:
+            raise ValueError(
```

Comment:
`VerifyException`? Test?

Reply:
Already is one

Reply:
Can separate it out if you feel it makes sense

Reply:
I mean we should probably have a filecheck test for this guy, instead of a test for the init.

Reply:
I'm struggling to trigger the error with filecheck as it hits a parsing error first, are you happy with this as is?

Reply:
Ah interesting, sure